.ci/deploy-package-registry.groovy

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env groovy
+
+@Library('apm@current') _
+
+pipeline {
+  agent { label 'ubuntu-20' }
+  environment {
+    BASE_DIR = "src/github.com/elastic/package-registry"
+    PIPELINE_LOG_LEVEL = 'INFO'
+    DOCKER_REGISTRY = 'docker.elastic.co'
+    DOCKER_REGISTRY_SECRET = 'secret/observability-team/ci/docker-registry/prod'
+    DOCKER_TAG = "${params.DOCKER_TAG}"
+    DOCKER_IMG_SOURCE = "${env.DOCKER_REGISTRY}/package-registry/distribution:production"
+    DOCKER_IMG_TARGET = "${env.DOCKER_REGISTRY}/package-registry/distribution:${env.DOCKER_TAG}"
+  }
+  options {
+    timeout(time: 1, unit: 'HOURS')
+    buildDiscarder(logRotator(numToKeepStr: '20', artifactNumToKeepStr: '20', daysToKeepStr: '30'))
+    timestamps()
+    ansiColor('xterm')
+    disableResume()
+    durabilityHint('PERFORMANCE_OPTIMIZED')
+    rateLimitBuilds(throttle: [count: 60, durationName: 'hour', userBoost: true])
+    quietPeriod(10)
+  }
+  parameters {
+    string(name: 'DOCKER_TAG', defaultValue: 'latest', description: 'The docker tag to be published.')
```

Comment:
Maybe we should define an allowed list of values? Otherwise, we may end up with typos or incorrect tags.

Reply:
This pipeline is intended to be consumed by the automation system. The automation system will be the one running the unified release process for staging and the git tag creation for the releases. 

Nevertheless, if someone runs this pipeline manually then there is a potential case of creating tags. Though, I'm not sure if this will be a case but a potential miss-use.

Otherwise, what values do you think should be verified? a regex with the major.minor.patch format?

Reply:
Yes, I was thinking about the semver regexp.

Reply:
Sure, let me add the regex validation. OTOH, there is one potential case by overriding an existing old release, for instance: `8.1.0`, it could be an accident if this job gets triggered, the automation is reliable since it's the one driving the official releases.

Reply:
> Sure, let me add the regex validation. OTOH, there is one potential case by overriding an existing old release

But the risk is present regardless of the presence of regexp, correct?

Reply:
Drive-by comment: in addition to providing guardrails against doing the wrong thing, this kind of validation also serves as a future reminder about the acceptable input, which may make maintenance easier. (Obviously, this can be accomplished via in-line documentation as well.)

Reply:
> But the risk is present regardless of the presence of regexp, correct?

Yes